server/src/main/java/org/opensearch/node/remotestore/RemoteStoreNodeService.java

```diff
@@ -183,6 +184,21 @@ public RepositoriesMetadata updateRepositoriesMetadata(DiscoveryNode joiningNode
                 boolean repositoryAlreadyPresent = false;
                 for (RepositoryMetadata existingRepositoryMetadata : existingRepositories.repositories()) {
                     if (newRepositoryMetadata.name().equals(existingRepositoryMetadata.name())) {
+                        try {
```

Comment:
Can the `try` block combined with the existing `try` in line#215

Reply:
I think its better to keep it separate for better readability? Also I don't think we should couple the skip flow with the execution/validation flow